internal/exec/oci_utils.go

```diff
@@ -1,101 +1,98 @@
-// https://opencontainers.org/
-// https://github.com/google/go-containerregistry
-// https://docs.aws.amazon.com/AmazonECR/latest/public/public-registries.html
-
 package exec
 
 import (
 	"fmt"
-	"io"
 	"os"
-	"path/filepath"
 
+	log "github.com/charmbracelet/log" // Charmbracelet structured logger
+
+	"github.com/google/go-containerregistry/pkg/authn"
 	"github.com/google/go-containerregistry/pkg/name"
 	"github.com/google/go-containerregistry/pkg/v1/remote"
-	"github.com/google/go-containerregistry/pkg/v1/tarball"
 	"github.com/google/uuid"
 
 	"github.com/cloudposse/atmos/pkg/schema"
-	u "github.com/cloudposse/atmos/pkg/utils"
 )
 
-// processOciImage downloads an Image from an OCI-compatible registry, extracts the layers from the tarball, and writes to the destination directory
+// Target artifact type for Atmos components
+const targetArtifactType = "application/vnd.oci.image.layer.v1.tar"
+
 func processOciImage(atmosConfig schema.AtmosConfiguration, imageName string, destDir string) error {
-	// Temp directory for the tarball files
+	// Temp directory for operations
 	tempDir, err := os.MkdirTemp("", uuid.New().String())
 	if err != nil {
 		return err
 	}
-
 	defer removeTempDir(atmosConfig, tempDir)
 
-	// Temp tarball file name
-	tempTarFileName := filepath.Join(tempDir, uuid.New().String()) + ".tar"
-
-	// Get the image reference from the OCI registry
+	// Parse image reference
 	ref, err := name.ParseReference(imageName)
 	if err != nil {
-		return fmt.Errorf("cannot parse reference of the image '%s'. Error: %v", imageName, err)
+		log.Error("Failed to parse OCI image reference", "image", imageName, "error", err)
+		return fmt.Errorf("invalid image reference: %v", err)
 	}
 
-	// Get the image descriptor
-	descriptor, err := remote.Get(ref)
-	if err != nil {
-		return fmt.Errorf("cannot get image '%s'. Error: %v", imageName, err)
+	// Authentication setup
+	githubToken := os.Getenv("GITHUB_TOKEN")
+	if githubToken == "" {
+		return fmt.Errorf("GITHUB_TOKEN environment variable is required")
 	}
+	auth := remote.WithAuth(&authn.Basic{
+		Username: "oauth2",
+		Password: githubToken,
+	})
 
-	// Download the image from the OCI registry
-	image, err := descriptor.Image()
+	// Get artifact descriptor
+	descriptor, err := remote.Get(ref, auth)
 	if err != nil {
-		return fmt.Errorf("cannot get a descriptor for the OCI image '%s'. Error: %v", imageName, err)
+		log.Error("Failed to get OCI image", "image", imageName, "error", err)
+		return fmt.Errorf("cannot get image '%s': %v", imageName, err)
 	}
-
-	// Write the image tarball to the temp directory
-	err = tarball.WriteToFile(tempTarFileName, ref, image)
+	// Check if we need to filter by MediaType
+	filterByArtifactType := descriptor.MediaType == targetArtifactType
```

Comment:
We should log warn if thr filterByArtigfactType is not the atmos component media type.